src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let resultSum = 0;
+
+  const funAdder = (a) => {
+    if (a === undefined) {
+      const returnValue = resultSum;
+
+      resultSum = 0;
+
+      return returnValue;
+    } else {
+      resultSum += a;
+
+      return funAdder;
```

Comment:
Returning the function itself for chaining calls is the key part of achieving the desired functionality. This is well implemented.